dist/api_data.py

```diff
@@ -708,6 +708,15 @@ def __cmp__(self, other):
 ]
 
 session_config = [
+    Config('cache_cursors', '', r'''
+        enable caching of cursors for reuse. Any calls to WT_CURSOR::close
+        for a cursor created in this session will mark the cursor
+        as cached and keep it available to be reused for later calls
+        to WT_SESSION::open_cursor. Cached cursors may be eventually
+        closed. This flag requires
+        ::wiredtiger_open or WT_CONNECTION::reconfigure to be called
+        with \c cache_cursors enabled.''',
```

Comment:
In the previous incarnation, I had one: the connection-wide flag was required in order to enable the session flag.  That made sense because there was some overhead to tracking/managing the caching. Now there isn't any extra overhead.  Would you want to see a connection-wide flag used in that same way, or rather a connection-wide flag that means "turn on caching in all sessions unless explicitly disabled"?

Reply:
A couple of points: the documentation string above still refers to a flag in `wiredtiger_open`.  I'd like less surface area rather than more, so suggest removing that from the docstring.

To the other point, I guess I'd ask why not have this behavior always enabled?  Is there any downside other than extra memory associated with session handles?

Reply:
@michaelcahill I'm open to enabling caching globally. I suppose I've been waiting to see if performance was comparable on a variety of use cases before taking that step.